plutus-scb/src/Plutus/SCB/App.hs

```diff
@@ -50,107 +61,115 @@ data Env =
-runAppClientM f wrapErr action =
-    App $ do
-        env <- asks f
-        result <- liftIO $ runClientM action env
-        case result of
-            Left err    -> throwError $ wrapErr err
-            Right value -> pure value
-
-runWalletClientM :: ClientM a -> App a
-runWalletClientM = runAppClientM walletClientEnv WalletClientError
-
-runNodeClientM :: ClientM a -> App a
-runNodeClientM = runAppClientM nodeClientEnv NodeClientError
-
-runSigningProcessM :: ClientM a -> App a
-runSigningProcessM = runAppClientM signingProcessEnv SigningProcessError
+type AppBackend m =
+        '[ GenRandomTx
+         , NodeFollowerEffect
+         , Error ServantError
+         , WalletEffect
+         , Error WalletAPIError
+         , Error ServantError
+         , NodeClientEffect
+         , Error ServantError
+         , SigningProcessEffect
+         , Error ServantError
+         , UUIDEffect
+         , ContractEffect
+         , ChainIndexEffect
+         , Error ServantError
+         , EventLogEffect ChainEvent
+         , Error SCBError
+         , Writer [Wallet.Emulator.Wallet.WalletEvent]
+         , Log
+         , Reader Connection
+         , Reader Env
+         , m
+         ]
+
+runAppBackend ::
+    forall m a.
+    ( MonadIO m
+    , MonadLogger m
+    , MonadUnliftIO m
+    )
+    => Env
+    -> Eff
+        '[ GenRandomTx
+         , NodeFollowerEffect
+         , Error ServantError
+         , WalletEffect
+         , Error WalletAPIError
+         , Error ServantError
+         , NodeClientEffect
+         , Error ServantError
+         , SigningProcessEffect
+         , Error ServantError
+         , UUIDEffect
+         , ContractEffect
+         , ChainIndexEffect
+         , Error ServantError
+         , EventLogEffect ChainEvent
+         , Error SCBError
+         , Writer [Wallet.Emulator.Wallet.WalletEvent]
+         , Log
+         , Reader Connection
+         , Reader Env
+         , m
+         ] a
+    -> m (Either SCBError a)
+runAppBackend e@Env{dbConnection, nodeClientEnv, walletClientEnv, signingProcessEnv, chainIndexEnv} =
+    runM
+    . runReader e
+    . runReader dbConnection
+    . runStderrLog
+    . writeToLog
+    . runError
+    . handleEventLog
+    . flip handleError (throwError . ChainIndexError)
+    . handleChainIndexClient chainIndexEnv
+    . handleContractEffectApp
+    . handleUUIDEffect
+    . flip handleError (throwError . SigningProcessError)
+    . SigningProcessClient.handleSigningProcessClient signingProcessEnv
+    . flip handleError (throwError . NodeClientError)
+    . handleNodeClientClient nodeClientEnv
+    . flip handleError (throwError . WalletClientError)
+    . flip handleError (throwError . WalletError)
+    . WalletClient.handleWalletClient walletClientEnv
+    . flip handleError (throwError . NodeClientError)
+    . handleNodeFollowerClient nodeClientEnv
+    . handleRandomTxClient nodeClientEnv
```

Comment:
yikes

Reply:
The wrapping of errors is a bit unfortunate :( I suppose I could use `Control.Monad.Freer.Extras.handleZoomedError`, although that wouldn't reduce the number of effects and it would only barely reduce the size of the definition of `runAppBackend`

Reply:
The size of these stack handlers doesn't scare me nearly as much as the thought that some of these effects might be sensitive to the order in which they're resolved.

Not sure what to do with that fear though...

Reply:
Yeah, I don't really mind the error handling so much since it's pretty obvious what it does.

Perhaps we just need a way of being clear about what each bit is doing. Maybe we should pick a convention like: if you have 3 conceptual blocks of handlers, then pull them out into a where clause with a type signature so it's clear what each of them does.

Reply:
I extracted some of the handlers into a where clause, but couldn't get the type signatures to be sufficiently general that I wouldn't have to spell out the entire remaining list each time (my attempt with constraints resulted in a warning about `effs` being ambiguous) so I resorted to `PartialTypeSignatures`: https://github.com/input-output-hk/plutus/pull/1895/commits/5ce568b59aeda4648d926eb63c413e9d0e3f59ca

Reply:
Nice!